pkg/tsdb/models.go

```diff
@@ -85,3 +88,88 @@ func NewTimeSeries(name string, points TimeSeriesPoints) *TimeSeries {
 		Points: points,
 	}
 }
+
+type DataFrames interface {
+	Encoded() ([][]byte, error)
+	Decoded() (data.Frames, error)
+}
+
+type decodedDataFrames struct {
+	decoded data.Frames
+	encoded [][]byte
+}
+
+func NewDecodedDataFrames(decodedFrames data.Frames) DataFrames {
+	return &decodedDataFrames{
+		decoded: decodedFrames,
+	}
+}
+
+func (ddf *decodedDataFrames) Encoded() ([][]byte, error) {
+	if len(ddf.encoded) == 0 {
```

Comment:
perhaps these should check for nil  and not length of zero? I'm not entirely sure, but perhaps alerting dataframes being nil and length of zero could mean different things.

Reply:
Good point. Had that first, but changed when investigating alerting test failure.